src/socket/address.cr

```diff
@@ -216,12 +216,24 @@ class Socket
       if addr = @addr4
         addr.s_addr == 0_u32
       elsif addr = @addr6
-        addr.__in6_u.__u6_addr8 == StaticArray[0_u8, 0_u8, 0_u8, 0_u8, 0_u8, 0_u8, 0_u8, 0_u8, 0_u8, 0_u8, 0_u8, 0_u8, 0_u8, 0_u8, 0_u8, 0_u8]
+        ipv6_addr8(addr) == StaticArray[0_u8, 0_u8, 0_u8, 0_u8, 0_u8, 0_u8, 0_u8, 0_u8, 0_u8, 0_u8, 0_u8, 0_u8, 0_u8, 0_u8, 0_u8, 0_u8]
       else
         raise "unreachable!"
       end
     end
 
+    private def ipv6_addr8(addr : LibC::In6Addr)
+      {% if flag?(:darwin) || flag?(:openbsd) || flag?(:freebsd) %}
+      addr.__u6_addr.__u6_addr8
+      {% elsif  flag?(:linux) && flag?(:musl) %}
```

Comment:
there's a double space here.
Also, can you indent the code in the macro conditions?